### DISCUSSION TITLE
Is it possible to migrate rabbitmq mnesia data when sever hostname changes ?

### DISCUSSION THREAD

**Discussion by xixiangzouyibian** [rabbitmq/rabbitmq-server#12289] [Other] [answered]
### Community Support Policy

- [X] I have read [RabbitMQ's Community Support Policy](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md)

### RabbitMQ version used

3.13.5

### How is RabbitMQ deployed?

RPM package

### Steps to reproduce the behavior in question

When server host name changes, the rabbitmq will be installed as new node name. For example, the sever host name changed from A to B. 
It looks the old node: A is no longer to use, and the new node B is created from scratch, which not including the data in the old node.
Do we have the way to migrate data from old node to the new one ?
BTW, `rabbitmqctl rename_cluster_name` is not meet expectation to change the node name, it only change the cluster name.

**Comment by michaelklishin** [comment, marked as answer]
You have several options:

 * [Override node's data directory location](https://www.rabbitmq.com/docs/relocate). This won't update queue identities in the database but you can re-declare the queues, and if all queues in the system are transient, you won't have anything to update
 * [Export and import definitions](https://www.rabbitmq.com/docs/definitions) and import them into newly started blank node. This won't migrate the messages but can be good enough
 * Finally, the most complex migration option which migrates everything is [Blue-Green deployment](https://www.rabbitmq.com/docs/blue-green-upgrade) (it assumes that the original node can be started)
 
 There used to be a command that updated the hostname part of internal queue identities but it was removed because it is not compatible with quorum queues and streams.

**Reply by xixiangzouyibian** [reply]
Got it. Thanks for comments.
